itest/litd_custom_channels_test.go

```diff
@@ -2046,6 +2051,176 @@ func testCustomChannelsLiquidityEdgeCases(_ context.Context,
 
 	logBalance(t.t, nodes, assetID, "after failed payment (asset "+
 		"invoice, strict forwarding)")
+
+	// Edge case: Check if the RFQ HTLC tracking accounts for cancelled
+	// htlcs. We achieve this by manually creating & using an RFQ quote with
+	// a set max amount. We first pay to a hodl invoice that we eventually
+	// cancel, then pay to a normal invoice which should succeed.
+
+	// We start by sloshing some funds in the Erin<->Fabia
+	sendAssetKeySendPayment(
+		t.t, erin, fabia, 100_000, assetID,
+		fn.None[int64](), lnrpc.Payment_SUCCEEDED,
+		fn.None[lnrpc.PaymentFailureReason](),
+	)
+	sendKeySendPayment(t.t, erin, fabia, 20_000)
+
+	logBalance(t.t, nodes, assetID, "balance after 1st slosh")
+
+	// We create the RFQ order. We set the max amt to ~180k sats which is
+	// going to evaluate to about 10k assets.
+	resQ, err := charlieTap.RfqClient.AddAssetSellOrder(
+		ctxb, &rfqrpc.AddAssetSellOrderRequest{
+			AssetSpecifier: &rfqrpc.AssetSpecifier{
+				Id: &rfqrpc.AssetSpecifier_AssetId{
+					AssetId: assetID,
+				},
+			},
+			PaymentMaxAmt:  180_000_000,
+			Expiry:         uint64(time.Now().Add(time.Hour).Unix()),
+			PeerPubKey:     dave.PubKey[:],
+			TimeoutSeconds: 100,
+		},
+	)
+	require.NoError(t.t, err)
+
+	// We now create a hodl invoice on Fabia, for 10k assets.
+	hodlInv := createAssetHodlInvoice(t.t, erin, fabia, 10_000, assetID)
+
+	// Charlie tries to pay via Dave, by providing the scid that was
+	// manually created above.
+	scid := rfqmsg.SerialisedScid(resQ.GetAcceptedQuote().Scid)
+	payInvoiceWithAssets(
+		t.t, charlie, dave, hodlInv.payReq, assetID, true,
+		fn.Some(lnrpc.Payment_IN_FLIGHT), fn.Some(scid),
+	)
+
+	time.Sleep(time.Second)
+
+	// We now assert that the expected numbers of HTLCs are present on each
+	// node.
+	// Reminder, Topology looks like this:
+	//
+	// Charlie <-> Dave <-> Erin <-> Fabia
+	//
+	// Therefore the routing nodes should have double the number of HTLCs
+	// required for the payment present.
+	assertNumHtlcs(t.t, charlie, 3)
+	assertNumHtlcs(t.t, dave, 6)
+	assertNumHtlcs(t.t, erin, 6)
+	assertNumHtlcs(t.t, fabia, 3)
+
+	// Now let's cancel the invoice on Fabia.
+	payHash := hodlInv.preimage.Hash()
+	_, err = fabia.InvoicesClient.CancelInvoice(
+		ctxb, &invoicesrpc.CancelInvoiceMsg{
+			PaymentHash: payHash[:],
+		},
+	)
+	require.NoError(t.t, err)
+
+	// There should be no HTLCs present on any channel.
+	assertNumHtlcs(t.t, charlie, 0)
+	assertNumHtlcs(t.t, dave, 0)
+	assertNumHtlcs(t.t, erin, 0)
+	assertNumHtlcs(t.t, fabia, 0)
+
+	// Now Fabia creates the normal invoice.
+	invoiceResp = createAssetInvoice(
+		t.t, erin, fabia, 10_000, assetID,
+	)
+
+	// Now Charlie pays the invoice, again by using the manually created
+	// scid. This payment should succeed.
+	payInvoiceWithAssets(t.t, charlie, dave, invoiceResp.PaymentRequest,
+		assetID, true, defaultPaymentStatusOpt, fn.Some(scid))
+
+	logBalance(t.t, nodes, assetID, "after manual rfq hodl")
```

Comment:
We don't actually create a HODL invoice here. So we only test adding HTLCs to the counter in the order manager. But we don't test that if we cancel a HODL invoice that we then can make a normal payment again, as the liquidity isn't used up yet.

Reply:
In the lines above we attempted to make a payment using multiple HTLCs (in the `assertNumHtlcs` calls above) using the same rfq scid

Since we asserted that HTLCs exist, then re-asserted that HTLCs do not exist anymore, that means the HTLCs got cancelled

am i missing smth?

Reply:
Hmm, I must've looked at an old version of the code or didn't see the `CancelInvoice` above. Looks good to me the way it is now.